platform-sdk/docs/proposals/DP-0x/0X - Tss block signing - Cryptography APIs.md

```diff
@@ -0,0 +1,498 @@
+2. **swirlds-native-support**: Gradle module that enables loading into memory compiled native libraries, so they can be used with JNI.
+3. **swirlds-cryptography-tss**: Gradle module for the TSS Library. This library-only client is the consensus node, so will remain close to it in `hedera-services` repository under `platform-sdk` folder.
+4. **swirlds-cryptography-signatures**: Gradle module for the Bilinear Pairings Signature Library.
+5. **swirlds-cryptography-pairings-api**: Gradle module for the Bilinear Pairings API. Minimizes the impact of adding or removing implementations.
+6. **swirlds-cryptography-alt128**: Gradle module that will provide the implementation for the Bilinear Pairings API using alt-128 elliptic curve. That curve has been chosen in accordance with EVM's support of it. Support for that curve will be provided by arkworks rust library. The module will include Java code, and rust code that will be compiled for all possible system architectures, and distributed in a jar under a predefined structure.
+
+
+### Libraries Specifications
+
+#### Swirlds Native support
+##### Overview
+Our implementation of the `swirlds-cryptography-pairings-api` will use native compiled libraries under the hood accessed with JNI.
+
+There are two possible ways of making the library accessible to Java code, so it can be accessed through JNI:
+1. Make it available in the classpath: The library would need to be previously installed in the executing environment and reference its location in the classpath env variable.
+2. Distribute it as an application dependency. The caveat here is that:
+
+    a. The distributed library needs to be compiled in every possible architecture the application is going to be executed on.
+
+    b. The library needs to be unpackaged from the jar as it cannot be accessed while it's compressed.
+
+
+The native-support library will help with the loading of the native library by implementing option 2.
+for this to work, the Java native libraries need to be included in the jar running the JNI wrapping code under the following structure:
+
+```
+ native-library-client.jar
+ |_WEB-INF
+    |_arch64
+      |_macos
+        |_libhedera_bls_jni.dylib
+    |amd64
+      |_macos
+        |_libhedera_bls_jni.dylib
+      |_linux
+        |_libhedera_bls_jni.so
+      |_windows
+        |_libhedera_bls_jni.dll
+    |x-86
+      |_linux
+        |_libhedera_bls_jni.so
+      |_windows
+        |_libhedera_bls_jni.dll
+  ...
+```
+
+lib folder is organized in subfolders by the platform identifier, as returned by `System.getProperty("os.arch")` and `System.getProperty("os.name")` 
+
+##### Constraints
+This module will not depend on hedera-services artifacts, so cannot include logging, metrics, configuration, or any other helper module from that repo.
```

Comment:
For Logging System.Logger can be used.

Reply:
Won't add that as the point is that it should not use swirlds-logging.
Either way the use cases for logging should be minimal or non-existent.
The users of this library should produce the logging.